Update muon shield to support latest Zürich designs, remove old code and cleanup (#678)
ShipSoft/FairShip: CHANGELOG.md

* **Corrections in MuonDIS simulation**
  The DIS interactions are now time-shifted to be consistent with the original incoming muon. Additionally, tracks from soft interactions of the original muon along with the muon's veto response are preserved (in muonDis.root) and included up to the DIS interaction point. To be noted that the muon veto points are manually added using add_muonresponse.py, which modifies the simulation file. This replaces the old method of "backward-travelling muon" to generate the incoming muon's veto response. All MuonDIS simulation scripts have been updated and consolidated within FairShip/muonDIS, ensuring consistency for new productions.
* Added a custom CrossSection branch to the simulation file to save the DIS cross sections from muonDIS.

> * New Optimized version for MS in geometry_config

Reviewer: It would be good to avoid non-standard acronyms (MS, HS etc. in the change log)

Reviewer: ```suggestion
* Added new optimised warm muon shield `warm_opt` in geometry_config
```

> * Added new parameters to MS definition

Reviewer: This is very vague. It might be worth mentioning which parameters, if users should know. Otherwise something like "Added new parameters to muon shield to support configurations from current optimisation campaign" would be clearer.

* **Corrections in MuonDIS simulation**
  The DIS interactions are now time-shifted to be consistent with the original incoming muon. Additionally, tracks from soft interactions of the original muon along with the muon's veto response are preserved (in muonDis.root) and included up to the DIS interaction point. To be noted that the muon veto points are manually added using add_muonresponse.py, which modifies the simulation file. This replaces the old method of "backward-travelling muon" to generate the incoming muon's veto response. All MuonDIS simulation scripts have been updated and consolidated within FairShip/muonDIS, ensuring consistency for new productions.
* Added a custom CrossSection branch to the simulation file to save the DIS cross sections from muonDIS.
* Added new warm muon shield `warm_opt` in geometry_config
* Added new parameters to muon shield to support configurations from current optimisation campaign
* Added event_inspector class to experimental analysis_toolkit to streamline usage of helper functions; Added dump_event() as a start.

> * Add (optional) MgB2 field map

Reviewer: Don't remove other change log entries, please!

* fix(geometry): fix typo of "vacuums" for strawtubes medium
* fix(reco): Fix segmentation fault due to dummy containers (#453, #519)
* fix: don't use TFile's deprecated attribute syntax

> * shipDet_conf behaviour no longer depends on the muon shield version.

Reviewer: Remove this line, it's already under the "changed" heading

* Set up of the shield name is now done using the `--shieldName` flag instead of `--scName`.
* Allow using standalone TPythia for use with ROOT 6.32+
* Use git-lfs to track ROOT files

> * AddressOf is outdated function in ROOT, changed with addressof (in field/)

Reviewer: This might be better under the "Fixed" heading?

Author: See new commit

> * ShipDet_conf do not depends anymore from MS version. 

Reviewer: ```suggestion
* shipDet_conf behaviour no longer depends on the muon shield version.
```

### Removed

* fix: Remove unused, unrunnable shipPatRec_prev.py
* feat(geometry): Dropped support for old geometries without DecayVolumeMedium explicitly set(pre 24.11 release case).
* Removed old options 7, 9, 10
* Removed cobalt option
* Removed stepGeo option

> * Removed the flag constant field in Absorber (HS) -> fixed to 1.7 T

Reviewer: It might be worth clarifying here that these all concern the muon shield.

* Set up of the shield name is now done using the `--shieldName` flag instead of `--scName`.
* Allow using standalone TPythia for use with ROOT 6.32+
* Use git-lfs to track ROOT files
* AddressOf is outdated function in ROOT, changed with addressof (in field/)
* shipDet_conf behaviour no longer depends on the muon shield version.
* feat(digi): Use STL vectors for SST digitisation

### Removed

* fix: Remove unused, unrunnable shipPatRec_prev.py
* feat(geometry): Dropped support for old geometries without DecayVolumeMedium explicitly set(pre 24.11 release case).
* MS: Removed old options 7, 9, 10
* MS: Removed cobalt option
* MS: Removed stepGeo option
* MS: Removed the flag constant field in Absorber (HS) -> fixed to 1.7 T

> * MS:  run_simScript.py: The --noSC flag is removed, whether or not a configuration is SC/hybrid depends on the config selected. Configurations are defined in the shield_db in geometry_config.py

Reviewer: ```suggestion
* MS: run_simScript.py: The --noSC flag is removed, whether or not a configuration is SC/hybrid depends on the config selected. Configurations are defined in the shield_db in geometry_config.py
```

* fix: Remove unused, unrunnable shipPatRec_prev.py
* feat(geometry): Dropped support for old geometries without DecayVolumeMedium explicitly set(pre 24.11 release case).

> * ShipMuonShield : Removed old option (7,9,10). Muonshield geometry is defined in geometry config.py and the SC is based on the choice made. Removed Cobalt option.

Reviewer: This is not particularly clear to anyone not familiar with the detailed implementation. I'd suggest splitting it up:
- Removed old options 7, 9, 10
- Removed cobalt option
- run_simScript.py: The --noSC flag is removed, whether or not a configuration is SC/hybrid depends on the config selected. Configurations are defined in the shield_db in geometry_config.py
